### DISCUSSION TITLE
Using promdump as Linux executable

### DISCUSSION THREAD

**Discussion by Flurb** [ihcsim/promdump#12] [Q&A]
Hi!

Is it possible to use promdump as a Linux executable? I do see a promdump executable in the file list of each release, but it seems to lack some functionality (like restoring, and it creates different directory names in the export).

And of course, I cannot use the kubectl-promdump executable in my Linux terminal.

Is there any way to do this?

**Comment by ihcsim** [comment]
The code of the promdump `core` only interfaces with the Prometheus library. Hence, the binary by itself only works with capturing dump and the metadata of the TDSB data blocks. Restore isn't a Prometheus function, hence, it's implemented in the kubectl plugin. Not ideal, but can you run the kubectl plugin in a Docker container? I added a [Dockerfile](https://github.com/ihcsim/promdump/blob/main/Dockerfile) recently.

**Reply by Flurb** [reply]
That's clear to me, but should creating a tarbal also work with the binary only? Something like:

./promdump -data-dir /data -min-time 0 > test.tar.gz 

Or is this only possible by using the full plugin?

**Reply by ihcsim** [reply]
That should be possible with just the promdump binary.

**Reply by Flurb** [reply]
I thought so as well, but the output is a bit weird (a 'al' directory instead of a 'wal' directory for example).

**Reply by ihcsim** [reply]
Sorry for the late reply. What is the output of `tar -tf test.tar.gz`? Can you add the `-debug` option to see if that generates any additional useful logs?
